internal/webhook/external/helpers.go

```diff
@@ -0,0 +1,70 @@
+package external
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/api"
+	"github.com/akuity/kargo/internal/indexer"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+type refreshResult struct {
+	totalWarehouses int
+	numFailures     int
+}
+
+func refresh(
+	ctx context.Context,
+	c client.Client,
+	l *logging.Logger,
+	repoName string,
+) (*refreshResult, error) {
+	var warehouses v1alpha1.WarehouseList
+	err := c.List(
+		ctx,
+		&warehouses,
+		client.MatchingFields{
+			indexer.WarehousesBySubscribedURLsField: repoName,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list warehouses: %w", err)
+	}
+
+	l.Debug("listed warehouses",
+		"num-warehouses", len(warehouses.Items),
```

Comment:
The message is already pretty clear that this is about Warehouses, so I don't think we need that context repeated in the fields.

```suggestion
		"count", len(warehouses.Items),
```